inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+    write(accnum_cln, file = acc_file_path, append = TRUE)
+    if (separate) {
+      write(paste0(dir_path, "/", accnum_cln, ".faa"),
+            file = "input.txt", append = TRUE)
+      write(paste0(">", accnum_cln),
+            file = paste0(accnum_cln, ".faa"), append = TRUE)
+      write(toString(seqs[accnum]),
+            file = paste0(accnum_cln, ".faa"), append = TRUE)
+    }
+  }
+  names(seqs) <- cln_names
+  writeXStringSet(seqs, sequences, format = "fasta")
+  return(length(seqs))
+}
+submit_full <- function(
+    dir = "/data/scratch",
+    DB = Sys.getenv("BLAST_DB", unset = "refseq"),
+    NHITS = Sys.getenv("BLAST_HITS", unset = 100),
+    EVAL = Sys.getenv("BLAST_EVALUE", unset = 0.00001),
+    sequences = "~/test.fa",
+    phylo = "FALSE",
+    by_domain = "FALSE",
+    domain_starting = "~/domain_seqs.fa",
+    type = "full",
+    job_code=NULL,
+    submitter_email=NULL,
+    advanced_options=NULL,
+    get_slurm_mails=FALSE
+) {
+  # Set working directory
+  setwd(dir)
+
+  advanced_options_names <- names(advanced_options[advanced_options == TRUE])
+
+  # Write job submission params to file
+  job_args <- list(
+    submission_type = type,
+    database = ifelse(phylo == FALSE, DB, NA),
+    nhits = ifelse(phylo == FALSE, NHITS, NA),
+    evalue = ifelse(phylo == FALSE, EVAL, NA),
+    submitter_email = submitter_email,
+    advanced_options = advanced_options_names,
+    job_code = job_code
+  )
+  yml <- yaml::as.yaml(job_args)
+  write(yml, "job_args.yml")
+
+  # Create a log file
+  write("START_DT\tSTOP_DT\tquery\tdblast\tacc2info\tdblast_cleanup\tacc2fa
+        \tblast_clust\tclust2table\tiprscan\tipr2lineage\tipr2da\tduration",
+        "logfile.tsv")
+
+  # Process sequences (local handling)
+  if (phylo == "FALSE") {
+    # Split the sequences if needed, store them locally
+    num_seqs <- get_sequences(sequences, dir_path = dir, separate = TRUE)
+
+    fasta <- Biostrings::readAAStringSet(sequences)
+    headers_original <- names(fasta)
+    headers_accnum <- names(fasta) |> purrr::map_chr(function(x) extractAccNum(x))
+
+    # Execute BLAST locally (instead of submitting jobs to the cluster)
+    for (i in 1:num_seqs) {
+      # Assume each sequence is saved separately
+      input_file <- paste0(dir, "/", headers_accnum[i], ".faa")
+      # output_file <- paste0(dir, "/blast_output_", i, ".txt")
+
+      # Construct the local BLAST command (make sure 'blastn' is available locally)
+      run_molevolvr_pipeline(input_file, DB, NHITS, EVAL, is_query = F, type, i)
+
+      #cmd <- sprintf(
+      #  "deltablast -query %s -db %s -out %s -num_alignments %d -evalue %f -remote",
+      #  input_file, DB, output_file, NHITS, EVAL
+      #)
+
+      # Execute BLAST locally
+      #system(cmd)
+
+      cat(sprintf("BLAST for sequence %d completed.\n", i), file=stderr())
+    }
+  } else {
+    # Handle phylogenetic analysis if needed
+    cat("Phylogenetic analysis is not supported in the local version yet.\n")
```

Comment:
Maybe I missed this, but why?

Reply:
I am still to add support for dblast, da and phylo. phylogeneti is supported but just for the full option. from the webapp it seems user can run just minimal analysis or full analysis.